### ISSUE TITLE
ExportImportDevice class doesn't fully correspond to twin format

### ISSUE THREAD

**Issue by koryshev** [Azure/azure-iot-sdk-java#298]
- **OS and version used:** Windows 10

- **Java runtime used:** java 1.8

- **SDK version used:** iot-service-client-1.14.0

# Description of the issue:
It appears that current DTO-class used for import operation doesn't fully correspond to device twin structure. During implementation of import operation I've run into the following issues:

- I could not import devices with the `selfSigned` authentication type since `ExportImportDevice.authentication` has the `thumbprint` field instead of `x509Thumbprint` and `AuthenticationMechanism.type` has `SELF_SIGNED` value instead of `selfSigned`.
- It is not possible to create an instance of `ExportImportDevice` containing empty `secondaryThumbprint` field but `RegistryManager.importDevices` doesn't require this field to be specified
- `ExportImportDevice` class doesn't allow to specify `tags` section

# Code sample exhibiting the issue:
N/A

# Console log of the issue:
N/A

# Need Support?
* Have a feature request for SDKs? Please post it on [User Voice](https://feedback.azure.com/forums/321918-azure-iot) to help us prioritize.
* Have a technical question? Ask on [Stack Overflow](https://stackoverflow.com/questions/tagged/azure-iot-hub) with tag “azure-iot-hub”
* Need Support? Every customer with an active Azure subscription has access to support with guaranteed response time.  Consider submitting a ticket and get assistance from Microsoft support team
* Found a bug? Please help us fix it by thoroughly documenting it and filing an issue on GitHub (C, Java, .NET, Node.js, Python).

**Comment by timtay-microsoft** [comment]
Thanks for filing the issues, we'll try to look into this soon

**Comment by JMayrbaeurl** [comment]
DTO classes like ExportImportDevice are not used for Json serialization/deserialization, but there are separate classes like ExportImportDeviceParser used for this. This special parser classes fully conform to the twin format specification, e.g. using x509Thumbprint instead of thumbprint,

Currently it's not fully clear to me, if you can use an empty secondaryThumbprint. I can't find the specification for that. Why do you think RegistryManager.importDevices doesn't require this field to be specified?

And yes you are right. Currently you can't specify tags and properties in the ExportImportDevice for importing devices. This is a missing feature in the current version of the SDK.

BTW. I've just provided a PR that reactivates the e2e test for importing and exporting devices. As soon as it gets merged in, you can start debugging the import/export approach of the SDK.

**Comment by koryshev** [comment]
> DTO classes like ExportImportDevice are not used for Json serialization/deserialization

Thank you, I didn't know that -- I've been guided by the [import sample](https://github.com/Azure/azure-iot-sdk-java/blob/master/service/iot-service-samples/device-manager-sample/src/main/java/samples/com/microsoft/azure/sdk/iot/DeviceManagerImportSample.java) and it doesn't use `ExportImportDeviceParser`, so probably it makes sense to update it.

As for not mandatory secondaryThumbprint, currently I'm using a custom class with empty secondaryThumbprint during the import and the `RegistryManager.importDevices` method successfully creates devices with empty value there:
```javascript
"x509Thumbprint": {
    "primaryThumbprint": "<primaryThumbprint>",
    "secondaryThumbprint": null
}
```

**Comment by JMayrbaeurl** [comment]
Using DTO classes in the import sample is perfectly fine. Probably they have originally used the wrong names in the DTO ExportImportDevice, e.g. for thumbprint for the X509Thumbprint field. But renaming them would introduce breaking changes.


I've checked the usage of empty secondaryThumbprint field, too. On adding a device in the Azure portal you're forced to enter a non-empty string for secondaryThumbprint. Otherwise the portal refuses to register the device. Therefore the current check in the SDK is doing the right thing.

For the missing 'tags' in Importing and Exporting devices with the SDK I've already prepared a fix that will enable this. I'll wait till @timtay-microsoft merges my latest PR, and then immediately provide a PR with the fix.

**Comment by timtay-microsoft** [comment]
The fix for this issue has been merged into master thanks to @JMayrbaeurl's PR. When we do our next release, I'll go ahead and close this issue. Feel free to try building from source and letting us know if this fix was satisfactory for you @koryshev

**Comment by timtay-microsoft** [comment]
The fix for this issue has been released as of Service Client 1.15.0 and Deps 0.7.0, so I'm closing this issue now.

**Comment by az-iot-builder-01** [comment]
@JMayrbaeurl, @koryshev, thank you for your contribution to our open-sourced project! Please help us improve by filling out this 2-minute [customer satisfaction survey](https://forms.office.com/Pages/ResponsePage.aspx?id=v4j5cvGGr0GRqy180BHbRx8jpG79OudPqY-BojcKjkJUMFFBQjgxNkw5ODJIRjBWTzI0Nk85R0pUQiQlQCN0PWcu)
